panel/io/model.py

```diff
@@ -192,22 +193,6 @@ def hold(doc: Document, policy: HoldPolicyType = 'combine', comm: Comm | None =
     comm: Comm
         The Comm to dispatch events on when the context manager exits.
     """
-    doc = doc or state.curdoc
-    if doc is None:
-        yield
-        return
-    held = doc.callbacks.hold_value
-    try:
-        if policy is None:
-            doc.unhold()
-        else:
-            doc.hold(policy)
-        yield
-    finally:
-        if held:
-            doc.callbacks._hold = held
-        else:
-            if comm is not None:
-                from .notebook import push
-                push(doc, comm)
-            doc.unhold()
+    deprecated('1.7.0', 'panel.io.model.hold', 'panel.io.document.hold')
```

Comment:
Currently, this is set up always to warn. Maybe we should change that as I feel this deprecation will be pretty verbose.

Reply:
I'd hope most people imported it as `from panel.io import hold`.

Reply:
Postponed the warning until 1.6.0